src/NuGet.Core/NuGet.ProjectModel/PackageSpecUtility.cs

```diff
@@ -70,20 +70,31 @@ public static bool IsSnapshotVersion(string version)
             // Update the framework appropriately
             targetFrameworkInfo.FrameworkName = GetFallbackFramework(
                 targetFrameworkInfo.FrameworkName,
+                false, // Imports is null, so doesn't really matter what we have here
+                null, // There are no imports here. This utility is used in pack ref scenarios
                 packageTargetFallback,
                 assetTargetFallback);
 
-            if (assetTargetFallback?.Any() == true)
+            var hasAssetTargetFallback = assetTargetFallback?.Any() == true;
+            var hasPackageTargetFallback = packageTargetFallback?.Any() == true;
+
+            if (hasAssetTargetFallback)
             {
                 // AssetTargetFallback
-                targetFrameworkInfo.AssetTargetFallback = assetTargetFallback.AsList();
+                // Legacy readers
+                targetFrameworkInfo.Imports = assetTargetFallback.AsList();
+                targetFrameworkInfo.AssetTargetFallback = true;
+                targetFrameworkInfo.AssetTargetFallbacks = assetTargetFallback.AsList();
```

Comment:
or better get list out of both `assetTargetFallback` and `packageTargetFallback` at the first pace and then use it through out